cirq/testing/equals_tester.py

```diff
@@ -61,8 +61,13 @@ def add_equality_group(self, *group_items: Any):
             assert hasattr(v1, '__eq__') == hasattr(v1, '__ne__')
             # Careful: python2 int doesn't have __eq__ or __ne__.
             if hasattr(v1, '__eq__'):
-                eq = v1.__eq__(v2)
-                ne = v1.__ne__(v2)
+                eq = NotImplemented
+                ne = NotImplemented
+                try:
+                    eq = v1.__eq__(v2)
+                    ne = v1.__ne__(v2)
+                except TypeError:
```

Comment:
This should never ever be happening! Don't ignore this error, it is an actual failure!